### ISSUE TITLE
Use-after-free on Game.stop: gamepad detach wrote through a freed buffer

### ISSUE THREAD

**Issue by sunarya-thito** [sunarya-thito/good#44]
Found by CI on its first run, and fixed in 570d4a7.

`InputRegistry.release` freed the raw input buffer and *then* called
`_gamepads?.detach()`. Detaching releases every occupied slot, and releasing a
slot publishes a button change through that buffer, so the write landed on freed
memory:

```
RangeError (length): Invalid value: Not in inclusive range 0..2: 1477908428
  TripleBuffer.beginWrite      (triple_buffer.dart:133)
  InputDevice._publish         (input_state.dart:452)
  InputDevice.setGamepadButton (input_state.dart:366)
  GamepadCollector.releaseSlot (gamepad.dart:155)
  GamepadCollector.detach      (gamepad.dart:91)
  InputRegistry.release        (input.dart:456)
  GameRuntime._disposeBuffers  (game.dart:2034)
  Game.stop                    (game.dart:828)
```

The slot index is read back out of the freed page, hence `1477908428`.

Ten tests in `packages/good/test/game_gamepad_test.dart` fail this way on Linux
and **all 475 pass on Windows**, where the freed page still reads plausibly. It
is not intermittent: it fires on every `Game.stop()` with a pad attached. The
local suite could not have caught it and neither could the analyzer.

The fix is ordering — detach while the storage is still alive, then dispose.

Kept as a record because it is the argument for a platform matrix: the reverse
case, something that only fails on Windows, is equally possible and equally
invisible today.
